server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+
+class RepoAssociate(View):
+    """
+    View to copy units between repositories.
+    """
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def post(self, request, dest_repo_id):
+        """
+        Associate units matching the criteria into the given repository
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param dest_repo_id: id of the repository content will be copied to
+        :type  dest_repo_id: str
+
+        :raises pulp_exceptions.MissingValue: if required param source_repo_id is not passed
+        :raises pulp_exceptions.InvalidValue: if source_repo_id is not found or if criteria
+                                              params cannot be parsed
+        :raises pulp_exceptions.OperationPostponed: dispatch a publish repo task
+        :raises pulp_exceptions.PulpDataException: if there is a problem parsing the criteria object
+        """
+
+        # Validate existence of dest_repo_id, can raise a MissingResource
+        repo_query_manager = manager_factory.repo_query_manager()
+        repo_query_manager.get_repository(dest_repo_id)
+
+        criteria_body = request.body_as_json.get('criteria', None)
+        overrides = request.body_as_json.get('override_config', None)
+        source_repo_id = request.body_as_json.get('source_repo_id', None)
+        if source_repo_id is None:
+            raise pulp_exceptions.MissingValue(['source_repo_id'])
+
+        # Catch MissingResource because this is body data, raise 400 rather than 404
+        try:
+            repo_query_manager.get_repository(source_repo_id)
+        except pulp_exceptions.MissingResource:
+            raise pulp_exceptions.InvalidValue(['source_repo_id'])
+
+        if criteria_body:
+            try:
+                criteria = UnitAssociationCriteria.from_client_input(criteria_body)
+            except pulp_exceptions.InvalidValue, e:
+                invalid_criteria = pulp_exceptions.InvalidValue('criteria')
+                invalid_criteria.add_child_exception(e)
+                raise invalid_criteria
+        else:
+            criteria = None
+
+        task_tags = [tags.resource_tag(tags.RESOURCE_REPOSITORY_TYPE, dest_repo_id),
+                     tags.resource_tag(tags.RESOURCE_REPOSITORY_TYPE, source_repo_id),
+                     tags.action_tag('associate')]
+        async_result = associate_from_repo.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, dest_repo_id, [source_repo_id, dest_repo_id],
+            {'criteria': criteria, 'import_config_override': overrides}, tags=task_tags)
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+
+class RepoUnassociate(View):
+    """
+    View to unassociate a unit from a repository.
+    """
+
+    @auth_required(authorization.UPDATE)
+    @json_body_allow_empty
```

Comment:
yes, it is a general criteria issue, i just checked consumer group unassociate members, which has same logic as repo unassociate units, and same issue http://fpaste.org/203243/73815491/
I think this is strange, because seems to me that previously it was working normally( i am not sure), maybe a regression bug?